Stasis Trap perk/scheme

robmart opened the pull request:
<!--
A basic changelog, goes to patch notes of the next version, so should be as short and informative as possible.
-->
## Changelog:
- Added stasis trap perk which unlocks stasis trap scheme:
    - Stuns character for three years, making them unable to use interactions

<!--
A changelog where you can describe more complicated things to other developers and testers.
-->
## Developer changelog:

## TODO: 
- ~~Event loc. Abduct loc: https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/pull/352#issuecomment-827965376~~ Made by the wonderful KingofMushan

<!--
If you need to explain something to testers. Otherwise, delete this part.
-->
# How to test:
- Try the new scheme:
    - From the actors perspective
    - From the targets perspective (can switch character right before excecuting the scheme as the actor)
    - When it succeeds
    - When it fails
- Make sure no interactions can be used when the character is stunned
- Feel free to tell me if the balance of the scheme is of (too easy/too difficult), or if the stunned modifier needs some sort of balancing

# Test Tips:
- With the `yesman` console command, everyone would like to join your plot.
- Use **Switch Character** from the Main Menu to change character your perspective to another character during the gameplay.

robmart committed:
Start of stasis trap - copied abduct

robmart committed:
Fixed minor errors and removed all events except path one

robmart committed:
Disable interactions when stunned pt1

robmart committed:
Disable interactions when stunned pt2

robmart committed:
Check now works

robmart committed:
Loc

robmart committed:
Game concept

robmart committed:
Change loc to use concept

robmart committed:
Loc

robmart committed:
More loc

robmart commented:
> 
> 
> Huh, that's not correct syntax, it's `has_trait`.

I meant to check if the character had modifier, messed it up somewhere and copy pasted it everywhere else.

robmart committed:
Perk name

zumbak04 commented on common/schemes/wc_stasis_trap_scheme.txt:
```diff
@@ -0,0 +1,293 @@
+		
+		is_adult = yes
+		is_landed = yes
+		is_imprisoned = no
```
You should check for the stunned trigger here.

zumbak04 commented on common/schemes/wc_stasis_trap_scheme.txt:
```diff
@@ -0,0 +1,293 @@
+			has_trait_with_flag = can_murder_own_children
+		}
+		scope:target = {
+			is_imprisoned = no
```
You should check for the stunned trigger here instead of `is_imprisoned`

zumbak04 commented on common/schemes/wc_stasis_trap_scheme.txt:
```diff
@@ -0,0 +1,293 @@
+	agent_join_chance = {
+		base = 0
+		
+		# Base Reluctance, including negative AI Energy
+		modifier = {
+			add = {
+				value = -20
+				if = {
+					limit = {
+						ai_energy < 0
+					}
+					add = {
+						value = ai_energy
+						multiply = 0.5
+					}
+				}
+			}
+			desc = BASE_RELUCTANCE
+		}
+
+		#Is Owner sending a gift in the invite interaction?
+		modifier = {
+			add = bribe_acceptance_value
+			exists = scope:gift
+			always = scope:gift
+			NOT = {
+				OR = {
+					scope:target = { is_parent_of = root }
+					root = { is_parent_of = scope:target }
+				}
+			}
+			desc = "SCHEME_WITH_GIFT"
+		}
```
I have no idea why Paradox didn't make it a scripted trigger, so please do.

zumbak04 commented on common/schemes/wc_stasis_trap_scheme.txt:
```diff
@@ -0,0 +1,293 @@
+		}
+		
+		# Various penalties
+		first_valid = {
+			modifier = { # Foreign Rulers are very hard to kidnap
+				add = -75
+				desc = "sway_foreign_target"
+				scope:target = {
+					is_ruler = yes
+					NOT = {
+						scope:target.top_liege = scope:owner.top_liege
+					}
+				}
+			}
+			modifier = { # Foreign subjects less so
+				add = -40
+				desc = "sway_foreign_target"
+				scope:target = {
+					is_foreign_court_or_pool_guest = no
+					NOT = {
+						scope:target.top_liege = scope:owner.top_liege
+					}
+				}
+			}
+			modifier = { # Rulers are harder to kidnap
+				add = -50
+				desc = "FABRICATE_HOOK_RULER_TARGET"
+				scope:target = {
+					is_ruler = yes
+					scope:target.top_liege = scope:owner.top_liege
+				}
+			}
+		}
```
Kidnap?

zumbak04 commented on common/trigger_localization/wc_character_interaction_triggers.txt:
```diff
@@ -0,0 +1,7 @@
+﻿stasis_trap_already_stunned = {
+	first_not = STASIS_TRAP_ALREADY_STUNNED
+}
+
+stasis_trap_you_are_stunned = {
+	first_not = STASIS_TRAP_YOU_ARE_STUNNED
+}
```
Check other trigger localizations.
`first` and `first_not` refer to a target in the first person (thus you) so there's no need in 2 trigger localizations.
`third` and `third_not` refer to a target in the third person (thus he, she).

Also, I propose you use this style in localization keys (instead of STASIS_TRAP_ALREADY_STUNNED):
`<Location Name>_<Type>`
Thus it's `stasis_trap_already_stunned_first`, `stasis_trap_already_stunned_first_not`, `stasis_trap_already_stunned_third`, etc.

zumbak04 commented on common/modifiers/wc_lifestyle_modifiers.txt:
```diff
@@ -5,4 +5,8 @@
 	health = 0.25
+}
+
+stasis_trap_stunned = {
```
Can't find a localization.

zumbak04 commented on events/scheme_events/stasis_trap_scheme/stasis_trap_outcome_events.txt:
```diff
@@ -0,0 +1,400 @@
+
+	immediate = {
+		show_as_tooltip = {
+			hard_imprison_character_effect = {
+				TARGET = scope:target
+				IMPRISONER = scope:owner
+			}
```
Imprison?

zumbak04 commented on events/scheme_events/stasis_trap_scheme/stasis_trap_outcome_events.txt:
```diff
@@ -0,0 +1,400 @@
+﻿namespace = stasis_trap_outcome
```
I suppose it's copypasted events so created scripted triggers, effects out of shared chunks of code.

robmart committed:
Merge branch 'physical-perks' into stasis-trap

robmart committed:
Stun trigger

robmart committed:
Use stun trigger

robmart committed:
Replace in prison triggers with stunned ones

robmart committed:
Stun not kidnap

robmart committed:
More script modifiers

robmart committed:
Use new effect instead of imprison one

robmart reviewed (commented):
Acording to Zum, the duration of the stun should be longer. Was thinking 3 years.

Edit: Done

robmart committed:
Merge branch 'physical-perks' into stasis-trap

robmart commented:
Abduct event loc, which the stasis trap loc should be based on
![bild](https://user-images.githubusercontent.com/4825809/116318302-d8e88100-a7b4-11eb-811f-494cc5d27ab0.png)
![bild](https://user-images.githubusercontent.com/4825809/116318324-dede6200-a7b4-11eb-8648-6d95190c8524.png)

robmart committed:
Stasis Trap setup loc

Co-authored-by: KingofMushan

robmart committed:
Merge branch 'physical-perks' into stasis-trap

robmart commented:
The method, "seized_roadside" should be changed to something else.

KingofMushan commented:
Wrote up these five dialogs for you!

Setup:
abduct_outcome.1003.t:0 "$abduct_event_title$ A Stunning Surprise"
 abduct_outcome.1003.desc:0 "With my agents now familiar with the most common places for my mark to frequent in [court_location.GetNameNoTierNoTooltip], [target.GetTitledFirstName] is within reach!\n\nMy agent assures me that they would be able to plant a particularly nasty trap just outside [target.GetTitledFirstNamePossessive] personal chambers. I know just the tool to use for this kind of hunt."
 abduct_outcome.1003.a:0 "Bring me my tools, I have a Stasis Trap to make."

Success:
abduct_outcome.2001.desc:0 "Success! The trap was placed discreetly and successfully, and [target.GetTitledFirstName] received quite the unexpected surprise. [target.GetSheHe] will likely be out of commission for quite some time..."
 abduct_outcome.2001.a:0 "I hope [target.GetSheHe] likes liquid food!"

 abduct_outcome.2002.desc:0 "Before [target.GetTitledFirstName] even knew what was happening, [target.GetSheHe] stepped quite literally into my trap! [target.GetSheHe|U] should be paralyzed for at least a few years."

 abduct_outcome.2003.desc:1 "I almost can’t believe it worked! It was as simple as my agent dropping one of my traps inside the [target.Custom('ResidenceBuilding')] in the middle of the night. \n\n[target.GetTitledFirstName] is no doubt kicking [target.GetHerselfHimself] over the poor security. Or at least, [target.GetSheHe] would be if [target.GetSheHe] could move [target.GetHerHis] legs to do so."

Failure:
abduct_outcome.4003.opening:0 "No! My agents were discovered trying to break into [target.GetTitledFirstNamePossessive] chambers. I should not have trusted those buffoons with such an important task."
 abduct_outcome.4003.owner_discovered:0 "\n\nWhat's worse, those fools confessed and gave up my involvement in the plot!"
 abduct_outcome.4003.owner_secret:0 "\n\nBy some miracle my allies made it out, and my involvement thankfully remains unknown."

Success (Target):
abduct_outcome.5001.t:0 "Stunning!"
 abduct_outcome.5001.opening.a:0 “Someone has struck me at my most vulnerable and left me paralyzed!”
 abduct_outcome.5001.opening.b:0 "I have foolishly fallen for a ploy by [owner.Custom2('MyCharacterIntroduction', ROOT.Char)] that has left me paralyzed!"
 abduct_outcome.5001.seized_roadside:0 "\n\nI was traveling with my retinue when I suddenly stepped on some sort of hunting trap. Before I knew it, I lost all control over my limbs and had to be carried back to [ROOT.Char.Custom(‘ResidenceBuilding’)] by my guards."
 abduct_outcome.5001.on_the_way:0 "\n\nI was on my way to town with just a handful of guards, and some sort of stasis trap was waiting right under my foot! By the time I regained consciousness, I had no control over my body."
 abduct_outcome.5001.unwelcome_guests:1 "\n\nA group of strangers must have broken into my chambers in the middle of the night and planted a trap! When I awoke in the morning, I stepped right onto it. It felt like I was struck by lightning! One of my courtiers found me hours later, completely paralyzed"
 abduct_outcome.5001.a:0 "Someone will pay for this..."

Failure (Target):
abduct_outcome.5002.t:0 "Close Call!"
 abduct_outcome.5002.seized_roadside:0 "I was traveling in [departure.GetNameNoTier] when one of my guards pointed out a hunting trap left precisely in my path! It was fortunate we noticed it when we did, for I dread to think what fate would’ve met me if I had stepped on it."
 abduct_outcome.5002.on_the_way:1 "I was on my way to town with only a handful of guards when they cleverly noticed a hunting trap set up just outside the gates. The trap was disarmed, and I was safely back in the [target.Custom('ResidenceBuilding')] before long."
 abduct_outcome.5002.unwelcome_guests:1 "A commotion woke me in the middle of the night. A group of thugs had been let into the [ROOT.Char.Custom('ResidenceBuilding')], and were only discovered once at the door to my chambers, attempting to set some sort of trap!"
 abduct_outcome.5002.fallback:0 "I have escaped a possible attempt on my life!"
 abduct_outcome.5002.owner_known:0 "\n\nThis is undoubtedly the work of none other than my [ROOT.Char.Custom2('RelationToMe', SCOPE.sC('owner'))], [owner.GetTitledFirstName]! Damn [owner.GetHerHim] to hell!"
 abduct_outcome.5002.owner_not_known:0 "\n\nI do not know who may be behind this deceitful attempt on my life, but they are still out there..."
 abduct_outcome.5002.a:0 "Oh, I’ll be ready for them next time!"
 abduct_outcome.5002.b:0 "I should give someone a raise for this."

robmart committed:
Event loc

robmart committed:
Wrong quotation marks

robmart committed:
Only triggers opinion mod if you know who did it

robmart committed:
One trigger loc instead of two

robmart committed:
Loc trigger now works properly

robmart committed:
Trigger loc

robmart committed:
All trigger loc works now

robmart committed:
Modifier loc

robmart committed:
Remove unnecessary code

robmart committed:
Trigger and opinion stuff

robmart committed:
Missed some stuff

robmart committed:
Renamed method to trapped_residence

zumbak04 commented on common/character_interactions/00_alliance.txt:
```diff
@@ -97,6 +97,14 @@
+
+		#Warcraft
+		scope:actor = { 
+			custom_description = {
+				text = stasis_trap_already_stunned
+				is_in_stun_trigger = no 
+			}
```
The custom description should be inside the scripted trigger.

zumbak04 commented on localization/english/wc_modifiers_l_english.yml:
```diff
@@ -4,3 +4,6 @@
  rebuilding_kingdom_modifier:0 "Rebuilding Kingdom"
+
+ stasis_trap_stunned:0 "Stunned!"
+ stasis_trap_stunned_desc:0 "You are [stunned|E]"
```
```suggestion
 stasis_trap_stunned_desc:0 "You are [stunned|E]."
```

zumbak04 commented on common/scripted_triggers/wc_scheme_triggers.txt:
```diff
@@ -0,0 +1,29 @@
+﻿dont_excecute_scheme_trigger = {
+	scope:scheme = {
+		add_scheme_progress = -10 # Reset scheme progress
+	}
+
+	stress_impact = {
+		impatient = medium_stress_impact_gain
+		stubborn = medium_stress_impact_gain
+	}
+}
+
+abandon_scheme_trigger = {
+	scope:scheme = {
+		end_scheme = yes
+	}
+
+	stress_impact = {
+		stubborn = medium_stress_impact_gain
+	}
+}
+
+restart_scheme_trigger = {
+	play_music_cue = "mx_cue_stress"
+	custom_tooltip = restart_scheme_tt
+	
+	scope:scheme = {
+		add_scheme_progress = -10 # Reset scheme progress
+	}
+}
```
Trigger? Check `error.log`. These are [Effects](https://ck3.paradoxwikis.com/effects), they go to scripted effects.

robmart replied:
Oops

zumbak04 commented on common/scripted_triggers/wc_health_triggers.txt:
```diff
@@ -0,0 +1,7 @@
+﻿is_in_stun_trigger = {
+	trigger_if = {
+		OR = {
+			has_character_modifier = stasis_trap_stunned
+		}
+	}
```
`trigger_if` should have `limit`. But I don't get why you use `trigger_if`.

robmart committed:
Effect not trigger

robmart committed:
Remove useless trigger_if

robmart committed:
Update localization/english/wc_modifiers_l_english.yml

Co-authored-by: zumbak04 <46384992+zumbak04@users.noreply.github.com>

robmart committed:
Trigger includes the loc

robmart committed:
All interactions uses correct loc

zumbak04 commented on common/character_interactions/00_character_interactions.txt:
```diff
@@ -38,6 +38,14 @@ offer_vassalization_interaction = {
 		}
+		
+		#Warcraft
+		custom_description = {
+			scope:actor = { 
+				text = stasis_trap_already_stunned
+				is_in_stun_trigger = no 
+			}
+		}
```
`custom_description` not inside the trigger.

zumbak04 commented on common/on_action/schemes/wc_stasis_trap_on_actions.txt:
```diff
@@ -0,0 +1,84 @@
+	}
+}
+
+stasis_trap_ongoing = {
```
I don't see any new events here so maybe you can remove it and refer to abduction or murder ongoing events?

zumbak04 commented on common/character_interactions/00_alliance.txt:
```diff
@@ -97,6 +97,11 @@
 				}
```
Errors:
```
[20:14:52][jomini_meantimetohappen.cpp:170]: unknown command 'hostile_scheme_base_reluctance' for MTTH in file common/schemes/wc_stasis_trap_scheme.txt line : 55
[20:14:52][jomini_dynamicdescription.cpp:66]: Unrecognized loc key stasis_trap_scheme_i_imprisoned_them_message.  file: common/schemes/wc_stasis_trap_scheme.txt line: 228
[20:14:52][pdx_persistent_reader.cpp:229]: Error: "Unknown trigger type: text, near line: 45
" in file: "common/character_interactions/00_character_interactions.txt" near line: 47
```
Also, these errors but they're from another branch.
```
[20:14:53][pdx_persistent_reader.cpp:229]: Error: "Unexpected token: wounded_chance_positive_modifier, near line: 8
" in file: " file: events/single_combat_events.txt line: 7760; calculate_injury_risks_effect" near line: 8
[20:14:53][pdx_persistent_reader.cpp:229]: Error: "Unexpected token: wounded_chance_positive_modifier, near line: 18
" in file: " file: events/single_combat_events.txt line: 7760; calculate_injury_risks_effect" near line: 18
[20:14:53][pdx_persistent_reader.cpp:229]: Error: "Unexpected token: wounded_chance_positive_modifier, near line: 8
" in file: " file: events/single_combat_events.txt line: 7864; calculate_injury_risks_effect" near line: 8
[20:14:53][pdx_persistent_reader.cpp:229]: Error: "Unexpected token: wounded_chance_positive_modifier, near line: 18
" in file: " file: events/single_combat_events.txt line: 7864; calculate_injury_risks_effect" near line: 18
[20:14:54][jomini_trigger.cpp:619]: Expected "is_shown = {", but got "is_shown = ".  file: common/decisions/wc_lifestyle_decisions.txt line: 35
```

zumbak04 replied:
So because it's based on the abduction, the final event doesn't mention if the scheme was discovered unlike other schemes like murder.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/117258236-cd085900-ae5d-11eb-86ce-f23271e04431.png)

</details>

zumbak04 replied:
No localization.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/117258701-4dc75500-ae5e-11eb-8c6e-93f5d3705bce.png)

</details>

Moreover, I shouldn't see this message because I just got the event telling me it's succeded. I don't need to see the same info twice.

zumbak04 commented on common/schemes/wc_stasis_trap_scheme.txt:
```diff
@@ -0,0 +1,269 @@
+				desc = "FABRICATE_HOOK_RULER_TARGET"
+				scope:target = {
+					is_ruler = yes
+					scope:target.top_liege = scope:owner.top_liege
+				}
+			}
+		}
+		
+		# At War
+		first_valid = {
+			modifier = {
+				add = -200
+				desc = "SCHEME_AT_WAR"
+				scope:target = {
+					is_at_war_with = scope:owner
+				}
+			}
+			modifier = {
+				add = -150
+				scope:target = {
+					is_landed = no
+					exists = liege
+					liege = {
+						is_at_war_with = scope:owner
+					}
+				}
+				desc = "SCHEME_AT_WAR_WITH_LIEGE"
+			}
+			modifier = {
+				add = -100
+				scope:target = {
+					is_landed = yes
+					exists = liege
+					liege = {
+						is_at_war_with = scope:owner
+					}
+				}
+				desc = "SCHEME_AT_WAR_WITH_LIEGE"
+			}
+		}
```
Reminder these are modifiers to balance abdication. It may be too harsh for this plot.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/116584985-fb031000-a928-11eb-90c2-0acd2afecb8c.png)

</details>

zumbak04 commented on events/scheme_events/stasis_trap_scheme/wc_stasis_trap_outcome_events.txt:
```diff
@@ -0,0 +1,326 @@
+#################
+
+######################
+# Seize Roadside
```
Is it still correct?

zumbak04 commented on common/scripted_modifiers/wc_hostile_scheme_scripted_modifiers.txt:
```diff
@@ -0,0 +1,34 @@
+﻿# Base Reluctance, including negative AI Energy
+hostile_scheme_base_reluctance_modifier = {
```
This modifier isn't used.

zumbak04 commented on common/scripted_effects/wc_scheme_effects.txt:
```diff
@@ -0,0 +1,183 @@
+
+		trigger_event = stasis_trap_outcome.5001
+
+		add_character_modifier = {
+			modifier = stasis_trap_stunned
+			years = 3
+		} 
```
Why `stun_character_effect` isn't used here?

zumbak04 replied:
Why you resolved it? I still don't get why there are 2 places that assign `stasis_trap_stunned`.

zumbak04 replied:
What I meant you should have used `stun_character_effect` here. Not remove the modifier adding from there.

zumbak04 commented on common/opinion_modifiers/wc_crime_and_prison_opinions.txt:
```diff
@@ -0,0 +1,11 @@
+	decaying = yes
+}
+
+attempted_stasis_trapion_opinion = {
+	opinion = -10
+	years = 5
+	decaying = yes
+}
```
So it's not a crime?

robmart committed:
Fix stun check

robmart committed:
stasis_trap_ongoing removed

robmart committed:
Misspelled modifier name

robmart committed:
Fixed errors

robmart committed:
Change comment

robmart committed:
Stunned modifier applied in the correct place

robmart committed:
Proper opinion modifiers

robmart committed:
Use murder modifiers

robmart committed:
2/3 of murder modifiers

robmart committed:
Merge branch 'physical-perks' into stasis-trap

robmart committed:
Merge branch 'physical-perks' into stasis-trap

robmart committed:
Merge branch 'physical-perks' into stasis-trap

zumbak04 commented on common/messages/wc_messagestxt:
```diff
@@ -0,0 +1,7 @@
+﻿stasis_trap_invalidated_i_stunned_message = {
```
Weird file name.

zumbak04 commented on common/schemes/wc_stasis_trap_scheme.txt:
```diff
@@ -0,0 +1,248 @@
+		}
+		
+		# At War
+		first_valid = {
+			modifier = {
+				add = -35
+				desc = "SCHEME_AT_WAR"
+				scope:target = {
+					is_at_war_with = scope:owner
+				}
+			}
+			modifier = {
+				add = -15
+				scope:target = {
+					is_landed = no
+					exists = liege
+					liege = {
+						is_at_war_with = scope:owner
+					}
+				}
+				desc = "SCHEME_AT_WAR_WITH_LIEGE"
+			}
+			modifier = {
+				add = -15
+				scope:target = {
+					is_landed = yes
+					exists = liege
+					liege = {
+						is_at_war_with = scope:owner
+					}
+				}
+				desc = "SCHEME_AT_WAR_WITH_LIEGE"
+			}
+		}
```
Can you make it a scripted modifier and pass the value of `-35` to it?

zumbak04 commented on common/schemes/wc_stasis_trap_scheme.txt:
```diff
@@ -0,0 +1,248 @@
+		}
+		
+		# Spiritual Head of Faith
+		modifier = {
+			add = -35
+			desc = "SCHEME_VS_SPIRITUAL_HOF"
+			scope:target = {
+				faith = scope:owner.faith
+				faith = {
+					has_doctrine = doctrine_spiritual_head
+				}
+				faith = {
+					exists = religious_head
+					religious_head = {
+			 			this = scope:target
+			 		}
+				}
+			}
+		}
```
Same, can you make it a scripted modifier?

zumbak04 commented on localization/english/wc_character_interactions_l_english.yml:
```diff
@@ -1,2 +1,7 @@
\ No newline at end of file
+ give_magic_eyes_interaction:0 "Give Magic Eyes"
+
+ stasis_trap_already_stunned_first_not:0 "I am not stunned"
+ stasis_trap_already_stunned_first:0 "I am stunned!"
+ stasis_trap_already_stunned_third_not:0 "[CHARACTER.GetShortUIName] is not stunned"
+ stasis_trap_already_stunned_third:0 "[CHARACTER.GetShortUIName] is already stunned"
```
And move it to `wc_character_triggers_l_english.yml`.
```suggestion
 stasis_trap_already_stunned_first_not:0 "I am #bold not#! stunned"
 stasis_trap_already_stunned_first:0 "I am stunned"
 stasis_trap_already_stunned_third_not:0 "[CHARACTER.GetShortUIName] is #bold not#! stunned"
 stasis_trap_already_stunned_third:0 "[CHARACTER.GetShortUIName] is already stunned"
```

zumbak04 commented on common/schemes/wc_stasis_trap_scheme.txt:
```diff
@@ -0,0 +1,248 @@
+		
+		is_adult = yes
+		is_landed = yes
+		is_imprisoned = no
```
So I would created basic triggers like `allow_schemes_trigger` including adult and imprisoned checks.

zumbak04 commented on common/schemes/wc_stasis_trap_scheme.txt:
```diff
@@ -0,0 +1,248 @@
+	
+	valid = {
+		is_landed = yes
+		OR = {
+			NOT = { is_parent_of = scope:target }
+			has_trait_with_flag = can_murder_own_children
+		}
+		scope:target = {
+			is_in_stun_trigger = no
+			exists = location
+			in_diplomatic_range = scope:owner 
+		}
+		# Special AI exceptions
+		NOT = {
+			scope:owner = {
+				is_ai = yes
+				has_opinion_modifier = {
+					target = scope:target
+					modifier = repentant_opinion
+				}
+			}
+		}
```
Make location check, AI exceptions, parent check a trigger called `valid_hostile_scheme_target_trigger`.

zumbak04 commented on common/character_interactions/wc_scheme_interactions.txt:
```diff
@@ -0,0 +1,110 @@
+	
+	is_shown = {
+		scope:actor = { has_perk = stasis_trap } # Scheme unlocked by the Stasis Trap Perk
+		NOR = {
+			scope:recipient = scope:actor
+		}
```
```suggestion
		NOT = { scope:recipient = scope:actor }
```

zumbak04 commented on common/character_interactions/wc_scheme_interactions.txt:
```diff
@@ -0,0 +1,110 @@
+		NOR = {
+			scope:recipient = scope:actor
+		}
+
+		#exists = scope:recipient.court_owner
+		
+		NAND = { # Game Rules!
+			has_game_rule = no_players_multiplayer_murder_schemes
+			scope:actor = {
+				is_ai = no
+			}
+			scope:recipient = {
+				is_ai = no
+			}
+		}
+		
+		NAND = {
+			has_game_rule = no_player_families_multiplayer_murder_schemes
+			scope:actor = {
+				is_ai = no
+			}
+			scope:recipient = {
+				OR = {
+					is_ai = no
+					any_close_family_member = {
+						AND = {
+							is_ai = no
+							NOT = { this = scope:actor }
+						}
+					}
+				}
+			}
+		}
```
So it's not a murder.

zumbak04 commented on common/opinion_modifiers/wc_crime_and_prison_opinions.txt:
```diff
@@ -0,0 +1,11 @@
+﻿stasis_traped_me_opinion = {
+	opinion = -30
+	years = 20
+	decaying = yes
+}
+
+attempted_stasis_trapion_opinion = {
+	opinion = -20
+	years = 10
+	decaying = yes
+}
```
It's still not a crime.
One player can do it against another player and not having a reason to punish.

robmart committed:
Update common/character_interactions/wc_scheme_interactions.txt

Co-authored-by: zumbak04 <46384992+zumbak04@users.noreply.github.com>

robmart committed:
Update localization/english/wc_character_interactions_l_english.yml

Co-authored-by: zumbak04 <46384992+zumbak04@users.noreply.github.com>

arithon committed:
started german translation, corrected type on wc_messages.txt file

robmart committed:
Remove game rule check

robmart committed:
allow_schemes_trigger

robmart committed:
modifier

robmart committed:
valid_hostile_scheme_target_trigger

robmart committed:
Trapping is a crime

robmart committed:
Rename file

robmart committed:
Merge branch 'physical-perks' into stasis-trap

robmart committed:
Merge branch 'physical-perks' into stasis-trap

robmart committed:
Merge branch 'physical-perks' into stasis-trap

zumbak04 commented on localization/english/wc_character_triggers_l_english.yml:
```diff
@@ -0,0 +1,7 @@
+﻿l_english:
+ give_magic_eyes_interaction:0 "Give Magic Eyes"
+
+ stasis_trap_already_stunned_first_not:0 "I am #bold not#! stunned"
+ stasis_trap_already_stunned_first:0 "I am stunned"
+ stasis_trap_already_stunned_third_not:0 "[CHARACTER.GetShortUIName] is #bold not#! stunned"
+ stasis_trap_already_stunned_third:0 "[CHARACTER.GetShortUIName] is already stunned"
```
Huh? New file? There's a file with the same name in `Warcraft-Guardians-of-Azeroth-2\localization\english\triggers`.

zumbak04 commented on localization/english/wc_character_interactions_l_english.yml:
```diff
@@ -1,2 +0,0 @@
-﻿l_english:
- give_magic_eyes_interaction:0 "Give Magic Eyes"
```
Huh, file deleted?

zumbak04 commented on common/scripted_modifiers/wc_scheme_scripted_modifiers.txt:
```diff
@@ -0,0 +1,54 @@
+			}
+		}
+
+	multiply = $MULTIPLIER$
```
Check `error.log` looking for `wc_`.
```
[19:18:36][jomini_meantimetohappen.cpp:392]: Unknown modifier 'multiply', referenced at  file: common/schemes/wc_stasis_trap_scheme.txt line: 61
```

zumbak04 commented on common/scripted_modifiers/wc_scheme_scripted_modifiers.txt:
```diff
@@ -0,0 +1,54 @@
+﻿scheme_base_success_chance_modifier = { #Use MULTIPLIER if you want harsher/softer penalties
```
Why you pass `MULTIPLIER`? It's much clearer to pass a value instead of `-35`.
Instead of `add = -15`, it will be:
```
add = {
	value = $value$
	multiply = 0.5
}
```

zumbak04 commented on common/scripted_triggers/wc_scheme_triggers.txt:
```diff
@@ -0,0 +1,26 @@
+}
+
+allow_schemes_trigger = {
+	AND = {
+		is_adult = yes
+		is_landed = yes
+		is_imprisoned = no
+		is_in_stun_trigger = no
+	}
```
`AND` isn't needed. By default, it returns true if all triggers are true.

zumbak04 commented on common/scripted_modifiers/wc_scheme_scripted_modifiers.txt:
```diff
@@ -0,0 +1,54 @@
+﻿scheme_base_success_chance_modifier = { #Use MULTIPLIER if you want harsher/softer penalties
+		# Spiritual Head of Faith
+		modifier = {
+			add = -35
+			desc = "SCHEME_VS_SPIRITUAL_HOF"
+			scope:target = {
+				faith = scope:owner.faith
+				faith = {
+					has_doctrine = doctrine_spiritual_head
+				}
+				faith = {
+					exists = religious_head
+					religious_head = {
+			 			this = scope:target
+			 		}
+				}
+			}
+		}
+		
+		# At War
```
Also, I would split these 2: `Spiritual Head of Faith` and `At War`. They differ in murder and abdication, so by uniting them you make these scripted modifiers unusable in these 2.

zumbak04 commented on localization/english/wc_modifiers_l_english.yml:
```diff
@@ -6,4 +6,7 @@
\ No newline at end of file
+ fortifying_feast_gain_modifier:0 "Satisfying Feast"
+
+ stasis_trap_stunned:0 "Stunned!"
```
No marks in names.
```suggestion
 stasis_trap_stunned:0 "Stunned"
```

zumbak04 commented on events/scheme_events/stasis_trap_scheme/wc_stasis_trap_outcome_events.txt:
```diff
@@ -0,0 +1,326 @@
+#####################################
+
+#######################
+# Seized Roadside
```
Roadsize? Like in abducation?

zumbak04 commented on localization/german/wc_character_interactions_l_german.yml:
```diff
@@ -0,0 +1,7 @@
+﻿l_german:
+ give_magic_eyes_interaction:0 "Give Magic Eyes"
+
+ stasis_trap_already_stunned_first_not:0 "Ich bin #bold nicht#! betäubt"
+ stasis_trap_already_stunned_first:0 "Ich bin betäubt"
+ stasis_trap_already_stunned_third_not:0 "[CHARACTER.GetShortUIName] ist #bold nicht#! betäubt"
+ stasis_trap_already_stunned_third:0 "[CHARACTER.GetShortUIName] ist bereits betäubt"
```
Note: these are moved to another file in English.

zumbak04 commented on localization/german/modifiers/wc_modifiers_l_german.yml:
```diff
@@ -29,3 +29,6 @@
  massive_infantry_toughness_mult:0 "$massive_infantry_toughness_add$"
  massive_infantry_screen_add:1 "[massive_infantry|E] [screen|E]"
+ 
+ stasis_trap_stunned:0 "Betäubt!"
```
```suggestion
 stasis_trap_stunned:0 "Betäubt"
```

zumbak04 commented on common/scripted_effects/wc_scheme_effects.txt:
```diff
@@ -0,0 +1,186 @@
+	
+	scope:target = {
+		add_character_flag = {
+			flag = was_stasis_traped_block_notification_event
```
I looked for this flag. It's never checked. Something is wrong compared to the vanilla?

zumbak04 commented on events/scheme_events/stasis_trap_scheme/wc_stasis_trap_outcome_events.txt:
```diff
@@ -0,0 +1,326 @@
+		first_valid = {
+			triggered_desc = {
+				trigger = {
+					scope:scheme_discovered = yes
```
I checked vanilla. Looks like it's an incorrect use.
Note: it's not the only wrong use in this file. Fix another as well.

robmart replied:
While I'm not going to resolve this conversation, I don't think this is a problem. Look at messages below.
![bild](https://user-images.githubusercontent.com/4825809/116904357-e29f3800-ac3d-11eb-9d61-cd4527b9b9e8.png)

robmart committed:
Update localization/english/wc_modifiers_l_english.yml

Co-authored-by: zumbak04 <46384992+zumbak04@users.noreply.github.com>

robmart committed:
Update localization/german/modifiers/wc_modifiers_l_german.yml

Co-authored-by: zumbak04 <46384992+zumbak04@users.noreply.github.com>

robmart committed:
Moved loc to correct files

robmart committed:
better modifiers

robmart committed:
Removed redundant AND

robmart committed:
Changed comment

robmart committed:
Removed unneeded flag

robmart committed:
Shouldn't apply stunned in two places

robmart committed:
Missed a equal sign

zumbak04 commented on common/lifestyle_perks/wc_physical_perks.txt:
```diff
@@ -145,7 +145,7 @@ physical_tree_2_perk_2 = {
 }
 
-physical_tree_2_perk_3 = {
+stasis_trap = {
```
The perk doesn't mention what it does.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/117035588-89a2d300-ad15-11eb-855c-d01590603795.png)

</details>

zumbak04 commented on common/lifestyle_perks/wc_physical_perks.txt:
```diff
@@ -145,7 +145,7 @@ physical_tree_2_perk_2 = {
 }
 
-physical_tree_2_perk_3 = {
+stasis_trap = {
```
The perk doesn't mention what it does.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/117035575-84458880-ad15-11eb-854d-9de10608e2b1.png)

</details>

robmart replied:
Resolving this because you duplicated it

zumbak04 commented on localization/english/wc_schemes_l_english.yml:
```diff
@@ -0,0 +1,50 @@
+ stasis_trap_action:0 "Stasis Trap"
+ stasis_trap_desc:0 "I will [stun|E] [SCHEME.GetTarget.GetShortUINameNoTooltip]!"
+ stasis_trap_desc_general:0 "Attempt to [stun|E] an enemy."
+ scheme_interaction_tt_stasis_trap_approved:0 "[recipient.GetFirstName] will soon be [stunned|E]!"
```
I don't approve lying in a tooltip.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/117035755-beaf2580-ad15-11eb-86a6-6d79ae45dd58.png)
![image](https://user-images.githubusercontent.com/46384992/117035783-ca9ae780-ad15-11eb-9f9f-ef34da38b59b.png)

</details>

robmart replied:
The english abduct equivalent is ` scheme_interaction_tt_abduct_approved:0 "[recipient.GetFirstName] will soon be my prisoner!"`

zumbak04 replied:
> The english abduct equivalent is ` scheme_interaction_tt_abduct_approved:0 "[recipient.GetFirstName] will soon be my prisoner!"`

It doesn't change my opinion.

zumbak04 commented on localization/english/wc_game_concepts_l_english.yml:
```diff
@@ -35,4 +35,10 @@
+ game_concept_wc_stunned:0 "Stunned"
+ game_concept_stunned:0 "Stunned"
+ game_concept_stun:0 "Stun"
+ game_concept_wc_stunned_desc:0 "A stunned character is unable to interract with any other character."
```
Like Character can't call alliances to give a player example of how to use it.

zumbak04 commented on localization/english/wc_schemes_l_english.yml:
```diff
@@ -0,0 +1,50 @@
+ stasis_trap_scheme_invalidated_title:1 "@scheme_icon! Stasis Trap Ends"
+ stasis_trap_invalidated_i_stunned_message:0 ""
+ stasis_trap_scheme_already_stunned_message:0 "[target.GetShortUIName|U] is already [stunned|E]"
+ stasis_trap_success_discovery_tt:0 "A successful trapping might reveal my identity\nOtherwise:"
```
Not sure why but looks like tabs are borken.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/117037490-96c0c180-ad17-11eb-9125-ffc8e5512c6a.png)

</details>

robmart replied:
The abduct one is the same
![bild](https://user-images.githubusercontent.com/4825809/117063012-7e0ad880-ad24-11eb-99b7-41ac8696ffa8.png)

robmart replied:
You never responded to this one, should it be changed or left as is?

zumbak04 replied:
> You never responded to this one, should it be changed or left as is?

I see. `stasis_trap_success_discovery_tt` doesn't play any role in this scheme unlike it was in the abduction scheme where it's clear who plotted against you once they finish.

zumbak04 replied:
Removed in 31e274acfaac6c4d809fee6f92037766b5461951.

zumbak04 commented on common/character_interactions/00_alliance.txt:
```diff
@@ -97,6 +97,11 @@
 				}
```
What about enabling animation once someone is stunned?
<details>
<summary>Click to expand</summary>

![Uploading image.png…]()

</details>

robmart replied:
Can't see image

zumbak04 replied:
> Can't see image

It doesn't play much role. The fact is stunned characters don't change animation to something more fitting.

robmart replied:
![bild](https://user-images.githubusercontent.com/4825809/117138229-0469fd80-adab-11eb-8352-4eac5469e669.png)

robmart replied:
Now uses the `still` animation when stunned, tell me if another would be better

robmart replied:
Now uses `pain` animation

robmart committed:
Merge branch 'physical-perks' into stasis-trap

zumbak04 commented on common/schemes/wc_stasis_trap_scheme.txt:
```diff
@@ -0,0 +1,181 @@
+	skill = intrigue
+	hostile = yes
+
+	power_per_skill_point = 2.0
+	resistance_per_skill_point = 1.75
+	power_per_agent_skill_point = 1
+	
+	minimum_progress_chance = 10
+	maximum_progress_chance = 90
+	minimum_success = 5
+	maximum_secrecy = 85
+	maximum_success = 85
```
I would consider changing these stats. Murder has a bigger maximum success and it's faster.

robmart replied:
Changed it to the murder ones, should maybe be even more lenient

zumbak04 replied:
> Changed it to the murder ones, should maybe be even more lenient

But its has less impactful result than murder. If its like murder, I would do murder instead.

robmart replied:
Changed them to be even more lenient, tell me if you'd like it to be even easier

zumbak04 replied:
> Changed them to be even more lenient, tell me if you'd like it to be even easier

 LGTM!

zumbak04 commented on common/character_interactions/00_alliance.txt:
```diff
@@ -97,6 +97,11 @@
 				}
```
It says I attempted murder.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/117038969-4a768100-ad19-11eb-8948-165a6efbfa6f.png)

</details>

robmart replied:
As far as I can tell all hostile schemes give the same modifier when discovered, check `hostile_scheme_discovery.3003`

zumbak04 replied:
> As far as I can tell all hostile schemes give the same modifier when discovered, check `hostile_scheme_discovery.3003`

Time to change it. You added new scheme after all.

zumbak04 commented on common/character_interactions/00_alliance.txt:
```diff
@@ -97,6 +97,11 @@
 				}
```
She's not, she ended my scheme by discovering I lead it.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/117039098-709c2100-ad19-11eb-9e80-cf56c0979a4c.png)

</details>

robmart replied:
Turns out i inverted the stasis check so everything that ended the scheme gave this message, should be fixed now

robmart committed:
Merge branch 'physical-perks' into stasis-trap

robmart committed:
Merge branch 'physical-perks' into stasis-trap

robmart committed:
Perk description

robmart committed:
Short desc of char interactions

robmart committed:
hostile scheme 3003 stasis trap desc

robmart committed:
Toast/message only appears when it should

robmart committed:
Use more lenient variables from the murder scheme

robmart commented:
@zumbak04 Can't reply to this review so I'll put it here
![bild](https://user-images.githubusercontent.com/4825809/117058155-94159a80-ad1e-11eb-98b8-776569486d6c.png)

Why do you want that? `stun_character_effect` is already called from `stasis_trap_outcome.5001` which is called from `stasis_trap_success_effect`, which means that it would change nothing. 

I did move the modifier applying to `stun_character_effect` though.

robmart committed:
Apply modifier in the correct spot

robmart committed:
Change loc

robmart committed:
Change scheme values

robmart committed:
Apply proper opinion modifiers

robmart committed:
Change animation when stunned

zumbak04 commented on .artist_resources/generic religion icons list.txt:
```diff
@@ -0,0 +1,114 @@
+## Vanilla
```
Mmmm. So to avoid such situations you shouldn't forget to update the destination branch of `physical-perks` and then merge the destination branch to this one.

robmart replied:
Lmao, resolved since this was your fault

robmart committed:
Using pain animation instead of still

robmart committed:
Moved it to the correct place

robmart committed:
Nvm shouldn't have moved it

zumbak04 commented on localization/english/wc_game_concepts_l_english.yml:
```diff
@@ -35,4 +35,14 @@
+ game_concept_wc_stunned:0 "Stunned"
+ game_concept_stunned:0 "Stunned"
+ game_concept_stun:0 "Stun"
+ game_concept_wc_stunned_desc:0 "A stunned character is unable to interract with any other character. Character interactions include, but are not limited to, declaring war, calling allies to war, using hooks, changing vassal contracts, and more."
```
Reminder. War, allies, hooks, vassal contacts are game concepts too and should be linked here as well instead of plain text.

zumbak04 committed:
Removed stasis_trap_success_discovery_tt

(GDC)

zumbak04 commented on localization/english/triggers/wc_character_triggers_l_english.yml:
```diff
@@ -14,4 +14,9 @@
\ No newline at end of file
+ THEIR_AGE_LESS_OR_EQUAL:0 "[CHARACTER.GetShortUIName|U] is $NUM|V0$ years old or under"
+
+ stasis_trap_already_stunned_first_not:0 "I am #bold not#! stunned"
+ stasis_trap_already_stunned_first:0 "I am stunned"
+ stasis_trap_already_stunned_third_not:0 "[CHARACTER.GetShortUIName] is #bold not#! stunned"
+ stasis_trap_already_stunned_third:0 "[CHARACTER.GetShortUIName] is already stunned"
```
Stunned should be a game concept here.
It's not clear why you can't use the decision.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/117260306-fcb86080-ae5f-11eb-8262-663272520caa.png)

</details>

zumbak04 commented on common/effect_localization/wc_perk_effects.txt:
```diff
@@ -8,4 +8,8 @@ blood_haze_perk_effect = {
 
```
I guess you would like to put the stunning check into this trigger to block activities like hunts.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/117260479-2d989580-ae60-11eb-825e-d8ba87a6622e.png)

</details>

zumbak04 commented on localization/english/wc_game_concepts_l_english.yml:
```diff
@@ -35,4 +35,14 @@
+
+ game_concept_stasis_trap:0 "Stasis Trap Scheme"
+ game_concept_stasis_trap_scheme:0 "Stasis Trap Scheme"
+ game_concept_stasis_trap_scheme_desc:0 "A [scheme|e] where a character attempts to [stun|e] another character for some time."
```
So make it clear from the beggining.
```suggestion
 game_concept_stasis_trap_scheme_desc:0 "A [hostile_scheme|E] where a character attempts to [stun|e] another character for some time blocking many interactions."
```

zumbak04 commented on localization/english/wc_game_concepts_l_english.yml:
```diff
@@ -35,4 +35,14 @@
+ game_concept_stun:0 "Stun"
+ game_concept_wc_stunned_desc:0 "A stunned character is unable to interract with any other character. Character interactions include, but are not limited to, declaring war, calling allies to war, using hooks, changing vassal contracts, and more."
+
+ game_concept_stasis_trap:0 "Stasis Trap Scheme"
+ game_concept_stasis_trap_scheme:0 "Stasis Trap Scheme"
```
Refers to their actual names.
```suggestion
 game_concept_stasis_trap:0 "$stasis_trap_action$"
 game_concept_stasis_trap_scheme:0 "$stasis_trap_scheme$"
```

zumbak04 commented on localization/english/wc_schemes_l_english.yml:
```diff
@@ -0,0 +1,52 @@
+ stasis_trap_action:0 "Stasis Trap"
+ stasis_trap_desc:0 "I will [stun|E] [SCHEME.GetTarget.GetShortUINameNoTooltip]!"
+ stasis_trap_desc_general:0 "Attempt to [stun|E] an enemy."
+ scheme_interaction_tt_stasis_trap_approved:0 "You will attempt to [stun|E] [recipient.GetFirstName]"
```
```suggestion
 scheme_interaction_tt_stasis_trap_approved:0 "You attempt to [stun|E] [recipient.GetFirstName]"
```

zumbak04 commented on localization/english/wc_schemes_l_english.yml:
```diff
@@ -0,0 +1,52 @@
+ stasis_trap_success_discovery_tt:0 "A successful trapping might reveal my identity\nOtherwise:"
+ do_not_execute_stasis_trap_tooltip:0 "I do not attempt the trapping"
+ 
+ STASIS_TRAP_SUCCESS_DESC:0 "You trap target, [stunning|E] them for a year"
```
```suggestion
 STASIS_TRAP_SUCCESS_DESC:0 "You trap the target, [stunning|E] them for 3 years"
```

zumbak04 commented on localization/english/wc_schemes_l_english.yml:
```diff
@@ -0,0 +1,52 @@
+ stasis_trap_scheme_desc:0 "[stun|E] your target"
+ stasis_trap_scheme_action:0 "Stasis Trap"
+ stasis_trap_action:0 "Stasis Trap"
+ stasis_trap_desc:0 "I will [stun|E] [SCHEME.GetTarget.GetShortUINameNoTooltip]!"
+ stasis_trap_desc_general:0 "Attempt to [stun|E] an enemy."
```
These are never used.

zumbak04 commented on localization/english/wc_schemes_l_english.yml:
```diff
@@ -0,0 +1,52 @@
+ stasis_trap_successful_roll_tt:0 "[target.GetShortUINameNoTooltip] becomes [stunned|E]!"
+ stasis_trap_become_discovered_roll_tt:0 "I am discovered!"
+ stasis_trap_scheme_invalidated_title:1 "@scheme_icon! Stasis Trap Ends"
+ stasis_trap_invalidated_i_stunned_message:0 ""
```
Why it's empty?

zumbak04 committed:
Removed stasis_trap_success_discovery_tt

(GDC)

zumbak04 commented on localization/english/wc_schemes_l_english.yml:
```diff
@@ -0,0 +1,51 @@
+
+ start_stasis_trap:0 "Stasis Trap #weak (Scheme)#!"
+ stasis_trap_scheme:0 "Stasis Trap Scheme"
+ stasis_trap_scheme_desc:0 "[stun|E] your target"
```
Because it sounded too plain.
```suggestion
 stasis_trap_scheme_desc:0 "I will [stun|E] your [SCHEME.GetTarget.GetShortUINameNoTooltip]."
```

robmart committed:
Update localization/english/wc_schemes_l_english.yml

Co-authored-by: zumbak04 <46384992+zumbak04@users.noreply.github.com>

robmart committed:
Update localization/english/wc_game_concepts_l_english.yml

Co-authored-by: zumbak04 <46384992+zumbak04@users.noreply.github.com>

robmart committed:
Update localization/english/wc_schemes_l_english.yml

Co-authored-by: zumbak04 <46384992+zumbak04@users.noreply.github.com>

robmart committed:
Update localization/english/wc_schemes_l_english.yml

Co-authored-by: zumbak04 <46384992+zumbak04@users.noreply.github.com>

robmart committed:
Applied Zumbaks suggestion

Co-authored-by: zumbak04 <46384992+zumbak04@users.noreply.github.com>

robmart committed:
Use game concepts in loc

robmart committed:
More game concept

robmart committed:
Can't be available if you are stunned

robmart committed:
Removed unused loc

robmart committed:
Added extra line which says if you are discovered or not

robmart committed:
Removed useless message and added loc

robmart committed:
Merge branch 'physical-perks' into stasis-trap

robmart committed:
Merge branch 'physical-perks' into stasis-trap

robmart committed:
Merge branch 'physical-perks' into stasis-trap

robmart committed:
Merge branch 'physical-perks' into stasis-trap

zumbak04 commented on common/scripted_effects/wc_scheme_effects.txt:
```diff
@@ -0,0 +1,238 @@
+﻿#Sets up the Outcome Roll values
+stasis_trap_outcome_roll_setup_effect = {
```
Is it me or every scheme has this effect unchanged? So there's no way to have it? You can just make one universal effect.

zumbak04 commented on common/scripted_effects/wc_scheme_effects.txt:
```diff
@@ -0,0 +1,238 @@
+
+#Roll to check whether a Stasis Trap Scheme is a success or a failure, and whether the owner is discovered
+stasis_trap_outcome_roll_effect = {
+	#SUCCESS ROLL
+	random = {
+		chance = scope:scheme.scheme_success_chance
+		custom_tooltip = stasis_trap_successful_roll_tt
+		save_scope_value_as = {
+			name = scheme_successful
+			value = yes
+		}
+	}
+	### End Success roll
+	#DISCOVERY ROLL
+	random = {
+		chance = scope:discovery_chance
+		custom_tooltip = stasis_trap_become_discovered_roll_tt
+		save_scope_value_as = {
+			name = scheme_discovered
+			value = yes
+		}
+	}
+	### End Discovery roll
```
Can be made into a universal effect.

zumbak04 commented on common/schemes/wc_stasis_trap_scheme.txt:
```diff
@@ -0,0 +1,162 @@
+
+	
+	on_invalidated = {
+		if = {
```
So the message still pops up though it's me who stunned him and I know he's stunned.
Also, it misses localization.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/117540832-12758380-b022-11eb-8e1c-1b6bd87612ac.png)

</details>

zumbak04 commented on common/scripted_effects/wc_scheme_effects.txt:
```diff
@@ -0,0 +1,238 @@
+
+		trigger_event = stasis_trap_outcome.5001
+	}
+
```
So the murder scheme gives stress reduction and I think it's nice.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/117540977-c5de7800-b022-11eb-9bd4-bfb8ef2fd2c5.png)

</details>

robmart committed:
Update localization/english/wc_schemes_l_english.yml

Co-authored-by: zumbak04 <46384992+zumbak04@users.noreply.github.com>

robmart committed:
Update common/scripted_triggers/00_available_for_events_triggers.txt

Co-authored-by: zumbak04 <46384992+zumbak04@users.noreply.github.com>

robmart committed:
Update events/scheme_events/stasis_trap_scheme/wc_stasis_trap_outcome_events.txt

Co-authored-by: zumbak04 <46384992+zumbak04@users.noreply.github.com>

robmart committed:
Update gfx/portraits/portrait_animations/animations.txt

Co-authored-by: zumbak04 <46384992+zumbak04@users.noreply.github.com>

robmart committed:
Update common/opinion_modifiers/wc_crime_and_prison_opinions.txt

Co-authored-by: zumbak04 <46384992+zumbak04@users.noreply.github.com>

robmart committed:
Message fix pt1

robmart committed:
Bring back the effect

robmart commented on common/effect_localization/wc_scheme_effects.txt:
```diff
@@ -0,0 +1,5 @@
+	global = stasis_trap_scheme_already_stunned_message
+	first = stasis_trap_scheme_already_stunned_message
+	third = stasis_trap_scheme_already_stunned_message
+}
```
Now uses correct scope, but the effect needs to be there for some reason.

robmart committed:
Close family gets negative opinion if target is successfully trapped

zumbak04 committed:
Past Effects

(GDC)

robmart committed:
Use hostile scheme effect in schemes

robmart committed:
Warcraft

robmart committed:
Merge remote-tracking branch 'origin/stasis-trap' into stasis-trap

robmart committed:
Better base secrecy

robmart committed:
Universal outcome roll

robmart committed:
Make success and discovery rolls into effects

robmart committed:
Stress reduction

robmart committed:
Use prowess bonus

robmart committed:
Merge branch 'physical-perks' into stasis-trap

zumbak04 committed:
hostile_scheme_opinion_effect

Fixed passing MURDERER instead of ATTACKER.
United attempted_hostile_scheme_opinion_effect and hostile_scheme_opinion_effect into 1 effect.

zumbak04 commented on common/opinion_modifiers/wc_crime_and_prison_opinions.txt:
```diff
@@ -0,0 +1,27 @@
+	imprisonment_reason = yes
+}
+
+stasis_traped_close_family_opinion = {
```
Has no localization.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/117568732-25945c00-b0d3-11eb-83e6-a96c5b997db7.png)

</details>

zumbak04 commented on common/game_concepts/wc_other_game_concepts.txt:
```diff
@@ -9,4 +9,12 @@ grim_religion = {
 	parent = religion
```
You're always discovered now.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/117568861-d26ed900-b0d3-11eb-8216-c5fd06413755.png)

</details>

zumbak04 commented on common/scripted_effects/00_murder_scheme_effects.txt:
```diff
@@ -262,9 +262,11 @@ murder_failure_effect = {
+			#Warcraft
+			attempted_hostile_scheme_opinion_effect = {
 				VICTIM = scope:target
 				MURDERER = scope:owner
```
You pass `MURDERER`, but it accepts `ATTACKER`. Another opinion effect has the same problem.

robmart replied:
Will be fixed when Zum merges `scheme-opinion` branch

zumbak04 replied:
Done!

zumbak04 committed:
Empty scope:scheme

Fixed passing empty scope:scheme since it ended before executing hostile_scheme_opinion_effect.

robmart committed:
Opinion loc

zumbak04 committed:
Merge pull request #368 from Warcraft-GoA-Development-Team/scheme-opinion

Scheme Opinion

robmart committed:
Fix discovery chance

robmart committed:
Merge remote-tracking branch 'origin/stasis-trap' into stasis-trap

arithon committed:
fixed some movement in german loc

zumbak04 commented on common/schemes/wc_stasis_trap_scheme.txt:
```diff
@@ -0,0 +1,175 @@
+﻿stasis_trap_scheme = {
```
It says I'm discovered though I'm not.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/117575175-b712c680-b0f1-11eb-8031-4d351a18f389.png)

</details>

robmart committed:
Missing loc

robmart committed:
Update common/schemes/wc_stasis_trap_scheme.txt

Co-authored-by: zumbak04 <46384992+zumbak04@users.noreply.github.com>

robmart committed:
Revert some effects to vanilla

robmart committed:
Merge remote-tracking branch 'origin/stasis-trap' into stasis-trap

zumbak04 committed:
Ending in Immediate

Paradox left comment in murder effect that ending scheme in immediate section of event is important and I would belive it.

arithon committed:
added 2 new files (german), added german translations, removed duplicated lines (english)

arithon committed:
Merge branch 'stasis-trap' of https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2 into stasis-trap

robmart committed:
Now uses correct discovered desc

robmart committed:
Merge remote-tracking branch 'origin/stasis-trap' into stasis-trap

robmart committed:
Merge branch 'physical-perks' into stasis-trap

robmart committed:
Merge branch 'physical-perks' into stasis-trap

zumbak04 commented on common/character_interactions/00_alliance.txt:
```diff
@@ -97,6 +97,11 @@
 				}
```
https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/pull/352#discussion_r628895781

So I analyzed this problem more. You created it here.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/117644589-af5b2c80-b19a-11eb-92eb-c502c77d5cb6.png)

</details>

Such a thing just doesn't exist in the vanilla. `scheme_discovered` is added only in this place and has only one value, `yes` or doesn't exist.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/117644779-ea5d6000-b19a-11eb-9c91-3b24ffb323b1.png)

</details>

Since you share many events with the vanilla, you may get more problems in the future if you don't replace `scheme_discovered` checking `yes` or `no`.

Also, you shouldn't pass `DISCOVERED` to `stasis_trap_success_effect`.

zumbak04 replied:
Also, I should notice that the originally stated problem is still present.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/117647146-a91a7f80-b19d-11eb-9f72-239dec7c9032.png)

</details>

Also, notice it's a number now.

zumbak04 commented on events/scheme_events/stasis_trap_scheme/wc_stasis_trap_outcome_events.txt:
```diff
@@ -0,0 +1,351 @@
+							exists = scope:scheme_discovered
+							scope:scheme_discovered = yes 
+						}
+						scope:scheme = {
+							is_scheme_exposed = yes
+						}
```
Exposed doesn't mean the owner is known. It means the target knows this scheme exists without implying the owner is known.

zumbak04 commented on common/scripted_effects/00_murder_scheme_effects.txt:
```diff
@@ -263,8 +263,8 @@ murder_failure_effect = {
 		}
 		show_as_tooltip = {
 			attempted_murder_opinion_effect = {
-				VICTIM = scope:target
-				MURDERER = scope:owner
```
You messed values here.

zumbak04 commented on events/scheme_events/murder_scheme/murder_outcome_events.txt:
```diff
@@ -4460,8 +4460,8 @@ murder_outcome.5002 = {
 				exists = scope:owner_to_reveal
 			}
 			attempted_murder_opinion_effect = {
-				VICTIM = root
-				MURDERER = scope:owner_to_reveal
```
You messed values here.

zumbak04 commented on common/scripted_effects/wc_scheme_effects.txt:
```diff
@@ -10,4 +10,357 @@
+	}
+}
+
+hostile_scheme_opinion_effect = {
```
So there's one problem and resolving will take a good amount of your time and I guess will be a good experience.

Basically, you can't pass `scope:scheme` to it because in many cases `scope:scheme` is over. Like in `stasis_trap_outcome.5001`.

zumbak04 replied:
> Basically, you can't pass `scope:scheme` to it because in many cases `scope:scheme` is over. Like in `stasis_trap_outcome.5001`.

Here's the event.

Of course, I haven't murdered anyone.

Steps to reproduce:
- I replaced this part to make sure it's always discovered.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/117648731-a28d0780-b19f-11eb-8de6-8835bb01928d.png)

</details>

- Enabled `yesman` so everyone joins my plot.
- Invited many trolls.
- Waited 11 months until the attempt event fired.
- Saved (for future tests).
- Switched to my target.
- Got the event.

zumbak04 replied:
I suppose you have to give up the idea of passing the scheme or make sure the scheme ends when all opinions are assigned.

zumbak04 commented on events/scheme_events/stasis_trap_scheme/wc_stasis_trap_outcome_events.txt:
```diff
@@ -0,0 +1,351 @@
+########################################
+
+#I have been stunned!
+stasis_trap_outcome.5001 = {
```
This text doesn't end with a dot.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/117648426-3f9b7080-b19f-11eb-832c-019ce0f1158e.png)

</details>

zumbak04 commented on events/scheme_events/stasis_trap_scheme/wc_stasis_trap_outcome_events.txt:
```diff
@@ -0,0 +1,351 @@
+		}
+	}
+
+	option = {
+		name = stasis_trap_outcome.5001.a
+		trigger = {
+			scope:scheme_discovered = yes
+		}
+		play_music_cue = "mx_cue_stress"
+		hostile_scheme_opinion_effect = {
+			VICTIM = scope:target
+			ATTACKER = scope:owner
+			SCHEME = scope:scheme
+		}
+	}
+
+	option = {
+		name = stasis_trap_outcome.5001.a
+		trigger = {
+			OR = {
+				scope:scheme_discovered = no
+				NOT = { exists = scope:scheme_discovered }
+			}
+		}
+		play_music_cue = "mx_cue_stress"
+	}
```
Both options are visible. I guess it's related to using `yes` or `no` I reported above.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/117648544-66f23d80-b19f-11eb-94c0-5444adf16356.png)

</details>

robmart committed:
Remove check

robmart committed:
Missing dot

robmart committed:
Fixed values

robmart committed:
Discovered in events now works properly

zumbak04 commented on common/scripted_effects/wc_scheme_effects.txt:
```diff
@@ -96,9 +96,14 @@ stasis_trap_success_effect = {
-			value = $DISCOVERED$
+		scope:scheme = {
+			if = {
+				limit = { exists = scope:scheme_discovered }
+			}
+			save_scope_value_as = {
+				name = discovered
+				value = yes
+			}
```
Why you set it? `scope:scheme_discovered` does the exactly same thing.

robmart replied:
Because you can't access the scheme_discovered scope in those events, the scheme is already ended at that point.

robmart committed:
Merge branch 'physical-perks' into stasis-trap

robmart committed:
Use scheme discovered

robmart committed:
Opinion now works

robmart committed:
Event now uses intrigue theme instead of prison

robmart committed:
Merge branch 'physical-perks' into stasis-trap

zumbak04 commented on events/scheme_events/stasis_trap_scheme/wc_stasis_trap_outcome_events.txt:
```diff
@@ -0,0 +1,333 @@
+			# If the Scheme Owner is known:
+			triggered_desc = {
+				trigger = { 
+					OR = {
+						exists = scope:scheme_discovered
+					}
```
You don't need OR if it's the only trigger.
```suggestion
					exists = scope:scheme_discovered
```

zumbak04 commented on events/scheme_events/stasis_trap_scheme/wc_stasis_trap_outcome_events.txt:
```diff
@@ -0,0 +1,333 @@
+		}
+	}
+	theme = intrigue
+	left_portrait = {
+		character = scope:owner
+		animation = scheme
+	}
```
You show the scheme owner here though they may be unknown. The issue comes from the abduction events where the owner is always known if the scheme is succesful.

robmart replied:
Solved in 
5975e49

zumbak04 commented on common/schemes/wc_stasis_trap_scheme.txt:
```diff
@@ -0,0 +1,166 @@
+	}
+
+	valid_agent = {
+		age >= 14
```
```suggestion
		age >= age_14_value
```

robmart committed:
Update events/scheme_events/stasis_trap_scheme/wc_stasis_trap_outcome_events.txt

Co-authored-by: zumbak04 <46384992+zumbak04@users.noreply.github.com>

zumbak04 commented on events/scheme_events/hostile_scheme_discovery_events.txt:
```diff
@@ -547,16 +559,20 @@ hostile_scheme_discovery.3003 = {
+		hostile_scheme_opinion_effect = {
+			VICTIM = scope:target
+			ATTACKER = scope:owner
+			SCHEME = scope:scheme.scheme_type
```
`scheme_type` can't be accessed like this.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/118822769-7cffac80-b8c9-11eb-8db8-b09ce7dbab5c.png)

</details>

If you have an idea how to fix it, tell it to me before implementing it.

robmart committed:
Update common/schemes/wc_stasis_trap_scheme.txt

Co-authored-by: zumbak04 <46384992+zumbak04@users.noreply.github.com>

robmart committed:
Portrait uses target if they don't know who did it

robmart committed:
Merge remote-tracking branch 'origin/stasis-trap' into stasis-trap

zumbak04 commented on common/scripted_triggers/wc_scheme_triggers.txt:
```diff
@@ -0,0 +1,24 @@
+﻿valid_hostile_scheme_target_trigger = {
+	is_landed = yes
```
So I can create a scheme targeting a landless character so I assume it doesn't work.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/118825186-8722aa80-b8cb-11eb-9901-d8734caa9df3.png)

</details>

zumbak04 commented on events/scheme_events/stasis_trap_scheme/wc_stasis_trap_outcome_events.txt:
```diff
@@ -234,6 +234,20 @@ stasis_trap_outcome.5001 = {
 			}
 		}
+
+		if = {
+			limit = {
+				exists = scope:scheme_discovered
+			}
+			scope:owner = {
+				save_scope_as = left_portrait
+			}
+		}
+		else = {
+			scope:target = {
+				save_scope_as = left_portrait
+			}
+		}
```
But the animation is still `scheme`. So showing the target "scheming" isn't correct.

robmart committed:
Opinion now works

robmart committed:
Stasis trap target can't be unlanded

zumbak04 committed:
hostile_scheme_opinion_effect Can Accept Schemes

hostile_scheme_opinion_effect can accept both schemes as objects and scheme types as strings.

zumbak04 committed:
Merge remote-tracking branch 'remotes/origin/master' into scheme-type

robmart committed:
Use proper animation

zumbak04 committed:
Merge remote-tracking branch 'remotes/origin/physical-perks' into stasis-trap

zumbak04 committed:
Merge remote-tracking branch 'remotes/origin/stasis-trap' into scheme-type

# Conflicts:
#	events/scheme_events/hostile_scheme_discovery_events.txt

robmart committed:
Merge pull request #384 from Warcraft-GoA-Development-Team/scheme-type

hostile_scheme_opinion_effect Can Accept Schemes

zumbak04 commented on common/scripted_triggers/wc_scheme_triggers.txt:
```diff
@@ -0,0 +1,25 @@
+﻿valid_hostile_scheme_target_trigger = {
+	is_landed = yes
```
I see you copied it to `scope:target`, but why you kept it here?

robmart committed:
Removed is_landed from wrong place

robmart committed:
Merge remote-tracking branch 'origin/stasis-trap' into stasis-trap

zumbak04 commented on common/scripted_modifiers/wc_scheme_scripted_modifiers.txt:
```diff
@@ -9,4 +9,69 @@
+scheme_hof_success_chance_modifier = {
+	# Spiritual Head of Faith
+	modifier = {
+		add = {
+			value = $value$
+		}
```
```suggestion
		add = $value$
```

zumbak04 commented on common/scripted_modifiers/wc_scheme_scripted_modifiers.txt:
```diff
@@ -9,4 +9,69 @@
+	# At War
+	first_valid = {
+		modifier = {
+			add = {
+				value = $value$
+			}
```
```suggestion
			add = $value$
```

zumbak04 commented on common/scripted_modifiers/wc_scheme_scripted_modifiers.txt:
```diff
@@ -9,4 +9,69 @@
+			desc = "SCHEME_AT_WAR_WITH_LIEGE"
+		}
+		modifier = {
+			add = {
+				value = $value$
+				multiply = 0.5
+			}
```
So it's harder to target a landed character.
```suggestion
			add = {
				value = $value$
				multiply = 0.75
			}
```

zumbak04 commented on events/scheme_events/stasis_trap_scheme/wc_stasis_trap_outcome_events.txt:
```diff
@@ -0,0 +1,356 @@
+			trigger = {
+				NOT = { exists = scope:scheme_discovered }
+			}
+			animation = anger
```
Because I don't really get why it's anger.
```suggestion
			animation = fear
```

zumbak04 commented on events/scheme_events/stasis_trap_scheme/wc_stasis_trap_outcome_events.txt:
```diff
@@ -0,0 +1,356 @@
+###################################
+
+#I was almost stunned
+stasis_trap_outcome.5002 = {
```
I guess this event's left character and its animations can be made like `stasis_trap_outcome.5001`.

robmart committed:
Apply suggestions from code review

Co-authored-by: zumbak04 <46384992+zumbak04@users.noreply.github.com>

robmart committed:
Update events/scheme_events/stasis_trap_scheme/wc_stasis_trap_outcome_events.txt

Co-authored-by: zumbak04 <46384992+zumbak04@users.noreply.github.com>

robmart committed:
5002 uses same animation/portrait as 5001

robmart commented on localization/english/wc_physical_lifestyle_l_english.yml:
```diff
@@ -39,6 +40,7 @@
  nightstalker_desc:0 "Grant a bonus to [scheme_hostile|E] [scheme_power|e] depending on your [prowess|e]"
  master_of_shadow_desc:0 "Successfully completing a [hostile_scheme|e] grants you the [GetModifier('master_of_shadows_modifier').GetNameWithTooltip] [modifier|e] \n(Lasts 10 years)"
  poison_blades_desc:0 "Your enemies are #N 10%#! more likely to take injuries."
+ stasis_trap_perk_desc:0 "Enables the [stasis_trap|E]"
```
The full name for the game concept is Stasis Trap Scheme, so your suggestion would make it Stasis Trap Scheme scheme.

arithon replied:
oh, my bad then. didn't check in game how it'd play out

robmart committed:
Update localization/english/wc_schemes_l_english.yml

Co-authored-by: arithon <47716948+arithon@users.noreply.github.com>

robmart committed:
Always Agents

robmart commented on localization/english/wc_schemes_l_english.yml:
```diff
@@ -0,0 +1,50 @@
+ start_stasis_trap_notification:0 "Stasis Trap"
+
+ stasis_trap_outcome.1001.t:0 "$stasis_trap_title$ A Stunning Surprise"
+ stasis_trap_outcome.1001.desc:0 "With my agents now familiar with the most common places for my mark to frequent in [court_location.GetNameNoTierNoTooltip], [target.GetTitledFirstName] is within reach!\n\nMy agent assures me that they would be able to plant a particularly nasty trap just outside [target.GetTitledFirstName]’s personal chambers. I know just the tool to use for this kind of hunt."
```
Fixed in 
e588d4d

robmart committed:
Merge branch 'physical-perks' into stasis-trap

robmart committed:
Merge branch 'physical-perks' into stasis-trap

zumbak04 commented on events/scheme_events/stasis_trap_scheme/wc_stasis_trap_outcome_events.txt:
```diff
@@ -0,0 +1,381 @@
+	}
+
+	immediate = {
+		if = {
+			limit = {
+				exists = scope:scheme_discovered
+			}
+			scope:owner = {
+				save_scope_as = revealed_owner
+			}
+		}
+
```
Why you kept it? It's not used anymore.

robmart replied:
Removed that part

Commit:
added French loc

arithon committed:
added final german file, finished physical lifestyle file (german)

arithon reviewed (approved):
english and german Loc LGTM

robmart committed:
Removed unused code

robmart committed:
Merge branch 'physical-perks' into stasis-trap

zumbak04 reviewed (approved):
LGTM!

robmart commented:
Finished, will merge when all perks are done.

zumbak04 commented:
> Finished, will merge when all perks are done.

I would say it needs the approval of another tester.

robmart commented:
> 
> 
> > Finished, will merge when all perks are done.
> 
> I would say it needs the approval of another tester.

Alright

TheGroxEmpire commented on localization/english/wc_modifiers_l_english.yml:
```diff
@@ -4,4 +4,6 @@
\ No newline at end of file
+
+ stasis_trap_stunned:0 "Stunned"
+ stasis_trap_stunned_desc:0 "You are [stunned|E]."
```
```suggestion
 stasis_trap_stunned_desc:0 "This character is [stunned|E]."
```

TheGroxEmpire reviewed (commented):
- The tooltip for Stunned modifier on another character says "You are stunned":
<details>

![image](https://user-images.githubusercontent.com/47816228/119371453-a4e37b80-bce0-11eb-85c0-ff65da71db2a.png)
</details>

- There are still plenty actions could be made with a stunned character i.e., decisions, council assignments, creating title, building. I'm not sure what's the limitation of stunned characters:
<details>

![image](https://user-images.githubusercontent.com/47816228/119372227-466acd00-bce1-11eb-9be5-7972415d2ffb.png)
![image](https://user-images.githubusercontent.com/47816228/119372396-774b0200-bce1-11eb-8f9b-9d1e6e4566f0.png)

</details>

- Personal events are still enabled. Are you intending on disabling this too while stunned? Could create a bit of odd scenario while being stunned:
<details>

![image](https://user-images.githubusercontent.com/47816228/119372689-c98c2300-bce1-11eb-9e19-732d6b336bb0.png)
</details>

- Minor UI bug:
<details>

![image](https://user-images.githubusercontent.com/47816228/119375018-7ebfda80-bce4-11eb-8d15-ce52f24da0d8.png)
</details>

- Suggestion: There should be an event popup after your character is no longer stunned. This minimally serves as a reminder that their character could do stuff that wasn't possible before.

- Suggestion: maybe it'd be nice to have extra plot chance for a scheme / imprisonment against a stunned character.

robmart commented:
> 
>     * There are still plenty actions could be made with a stunned character i.e., decisions, council assignments, creating title, building. I'm not sure what's the limitation of stunned characters:
> 
> 
>     * Personal events are still enabled. Are you intending on disabling this too while stunned? Could create a bit of odd scenario while being stunned:
> 
> 

Zumbak and I decided that if we disabled everything then that is going to make the stunned modifier unbearably boring for the player. Disabling interactions is good enough.

robmart commented:
> 
>     * Suggestion: There should be an event popup after your character is no longer stunned. This minimally serves as a reminder that their character could do stuff that wasn't possible before.
> 
>     * Suggestion: maybe it'd be nice to have extra plot chance for a scheme / imprisonment against a stunned character.

Both suggestions are good, will implement

robmart committed:
Update localization/english/wc_modifiers_l_english.yml

Co-authored-by: The Grox Empire <47816228+TheGroxEmpire@users.noreply.github.com>

robmart committed:
Already check in interaction, don't need it here

robmart committed:
Merge remote-tracking branch 'origin/stasis-trap' into stasis-trap

robmart committed:
Stasis trap worn off event

robmart committed:
Stunned target gives small boost to scheme power

TheGroxEmpire reviewed (commented):
The UI bug seems to be fixed and my suggestion were incorporated accordingly.
Just one small fix needed.
- The animation for the event ``stasis_trap_outcome.5003`` could be changed into something more neutral like standing upright, etc:
<details>

![image](https://user-images.githubusercontent.com/47816228/119445227-aea7c600-bd56-11eb-8b17-b6d0c5228263.png)
</details>

robmart committed:
Anger animation instead

robmart committed:
Stunned disables combat leadership

TheGroxEmpire reviewed (approved):
LGTM!

zumbak04 commented on localization/english/wc_modifiers_l_english.yml:
```diff
@@ -4,4 +4,6 @@
- 
\ No newline at end of file
+
+ stasis_trap_stunned:0 "Stunned"
```
The name of the modifier stopped showing.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/120220112-4de42600-c24d-11eb-88ac-5c62b9cc212c.png)

</details>

zumbak04 commented on common/schemes/wc_stasis_trap_scheme.txt:
```diff
@@ -26,9 +26,6 @@
 		valid_hostile_scheme_target_trigger = yes
-		scope:target = {
-			is_in_stun_trigger = no
-		}
```
The comment is so wrong. If you have it in the interaction, it doesn't mean you don't need it here.

Why? Because it made the scheme invalid if someone else stunned this character.

zumbak04 commented on common/modifiers/wc_lifestyle_modifiers.txt:
```diff
@@ -24,4 +24,10 @@ master_of_shadows_modifier = {
+stasis_trap_stunned = {
+	icon = intrigue_negative
+
+	disables_combat_leadership = yes
```
It doesn't work.
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/120221302-39089200-c24f-11eb-980e-0e4e74e1cb26.png)

</details>

Commit:
french loc updated

robmart committed:
Merge branch 'physical-perks' into stasis-trap

robmart committed:
Merge remote-tracking branch 'origin/stasis-trap' into stasis-trap

robmart committed:
Merge branch 'physical-perks' into stasis-trap

robmart committed:
Merge branch 'physical-perks' into stasis-trap

robmart commented:
It looks like scheme progress in 1.4.0 has a max of 10 instead of 100.

robmart committed:
Merge branch 'physical-perks' into stasis-trap

robmart committed:
Changed where the stun trigger is

robmart committed:
Doesn't work

robmart commented:
Need to disable combat leadership somehow

robmart committed:
Merge branch 'physical-perks' into stasis-trap

robmart committed:
Can't lead armies when stunned

zumbak04 reviewed (approved):
LGTM!

arithon committed:
translation of new lines (german) and one fix in english loc

arithon reviewed (approved):
loc LGTM

zumbak04 committed:
Moving Scheme Scripted Things

(GDC)
